File: src/main/java/ru/pgk/pgk/features/teacher/service/queries/TeacherQueriesService.java
package ru.pgk.pgk.features.teacher.service.queries;

import org.springframework.data.domain.Page;
import ru.pgk.pgk.features.teacher.entities.TeacherEntity;

public interface TeacherQueriesService {

    Page<TeacherEntity> getAll(String name, Integer offset);

    TeacherEntity getById(Integer id);

    TeacherEntity getByCabinet(String cabinet);
}

File: teacher-service/src/main/kotlin/ru/pgk/teacher_service/features/services/queries/TeacherQueriesService.kt
package ru.pgk.teacher_service.features.services.queries

import org.springframework.data.domain.Page
import ru.pgk.teacher_service.features.entities.TeacherEntity

interface TeacherQueriesService {
    fun getAll(name: String?, offset: Int): Page<TeacherEntity>
    fun getById(id: Int): TeacherEntity
    fun getByCabinet(cabinet: String): TeacherEntity
}